cryptol-style.md

```diff
@@ -0,0 +1,405 @@
+
+# STOPPED !!!
+
+Comments
+--------
+
+### Punctuation
+
+Write proper sentences; start with a capital letter and use proper
+punctuation.
+
+### Top-Level Definitions
+
+Comment every top level function (particularly exported functions),
+and provide a type signature; use Haddock syntax in the comments.
+Comment every exported data type. Function example:
+
+```haskell
+-- | Send a message on a socket. The socket must be in a connected
+-- state. Returns the number of bytes sent. Applications are
+-- responsible for ensuring that all data has been sent.
+send :: Socket      -- ^ Connected socket
+     -> ByteString  -- ^ Data to send
+     -> IO Int      -- ^ Bytes sent
+```
+
+For functions the documentation should give enough information to
+apply the function without looking at the function's definition.
+
+Record example:
+
+```haskell
+-- | Bla bla bla.
+data Person = Person
+    { age  :: !Int     -- ^ Age
+    , name :: !String  -- ^ First name
+    }
+```
+
+For fields that require longer comments format them like so:
+
+```haskell
+data Record = Record
+    { -- | This is a very very very long comment that is split over
+      -- multiple lines.
+      field1 :: !Text
+
+      -- | This is a second very very very long comment that is split
+      -- over multiple lines.
+    , field2 :: !Int
+    }
+```
+
+### End-of-Line Comments
+
+Separate end-of-line comments from the code using 2 spaces. Align
+comments for data type definitions. Some examples:
+
+```haskell
+data Parser = Parser
+    !Int         -- Current position
+    !ByteString  -- Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
+### Links
+
+Use in-line links economically. You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
+if:
+
+* The user might actually want to click on it for more information (in
+  your judgment), and
+
+* Only for the first occurrence of each API name in the comment (don't
+  bother repeating a link)
+
+Naming
+------
+
+Use camel case (e.g. `functionName`) when naming functions and upper
+camel case (e.g. `DataType`) when naming data types.
+
+For readability reasons, don't capitalize all letters when using an
+abbreviation. For example, write `HttpServer` instead of
+`HTTPServer`. Exception: Two letter abbreviations, e.g. `IO`.
+
+### Modules
+
+Use singular when naming modules e.g. use `Data.Map` and
+`Data.ByteString.Internal` instead of `Data.Maps` and
+`Data.ByteString.Internals`.
+
+Dealing with laziness
+---------------------
+
+By default, use strict data types and lazy functions.
+
+### Data types
+
+Constructor fields should be strict, unless there's an explicit reason
+to make them lazy. This avoids many common pitfalls caused by too much
+laziness and reduces the number of brain cycles the programmer has to
```

Comment:
In this context, "strict" and "point-free style" are new terms to me.  Are they important at this level?